facebook/lexical
# [lexical][auto-link] Fix auto link node escapes on second "."

## Description
The period in the separators list `PUNCTUATION_OR_SPACE ` leads to an auto-link creation prematurely before the user completes typing. Removing the "." from the list. 
Inspired by https://github.com/facebook/lexical/pull/5275 which was reverted because of Regexp changes.

**Closes:** #5254 #3546

## Test plan
1. Open https://playground.lexical.dev/
2. Type "https://www.foo.com"
3. See that the link is created for the full URL.

### Before

https://github.com/facebook/lexical/assets/5062807/032f41a5-fd2c-441b-81ba-fbba7aa3aa10



### After

https://github.com/facebook/lexical/assets/5062807/92d2f557-2a45-4954-857a-88a4d428e4b6
